packages/components/src/components/rating-stars/rating-stars.tsx

```diff
@@ -9,298 +9,195 @@
-  element: HTMLElement;
-
-  @Element() hostElement: HTMLElement;
-  /** (optional) hoverValue  */
-  @State() hoverValue = 0;
-  /** (optional) hoverValue  */
-  @State() isHovering = false;
-  /** (optional) max  */
-  @Prop() max = 5;
-  /** (optional) value  */
-  @Prop({ mutable: true }) value = 0;
-  /** (optional) small  */
-  @Prop() small = false;
-  /** (optional) disabled  */
-  @Prop() disabled = false;
-  /** (optional) ariaLabelTranslation  */
-  @Prop({ mutable: true })
-  ariaLabelTranslation = `${this.value} out of ${this.max} stars`;
-  /** (optional) precision  */
-  @Prop() precision = 1;
-  /** (optional) slider label */
-  @Prop() label?: string;
-  /** Emitted when the value has changed. */
-  @Event() scaleRatingChange!: EventEmitter<{ value: number }>;
-
-  colorFilled = `var(--scl-color-primary)`;
-  colorBlank = `var(--scl-color-grey-50)`;
-  size = this.small ? '16px' : '24px';
+  @Element() host: HTMLElement;
+
+  ratingStarId = `scale-rating-star-${ratingStarCount++}`;
+
+  /** The lower limit of the rating. In cases where  */
+  @Prop({ reflect: true }) starSize: 'small' | 'large' = 'large';
+  /** The lower limit of the rating. In cases where  */
+  @Prop({ reflect: true }) minRating = 0;
+  /** The upper limit of the rating */
+  @Prop({ reflect: true }) maxRating = 5;
+  /** Represents the current value of the rating */
+  @Prop({ mutable: true, reflect: true }) rating = 0;
+  /** makes the rating non-interactive (but still accessible)  */
+  @Prop({ reflect: true }) readonly = false;
+  /** disables input  */
+  @Prop({ reflect: true }) disabled = false;
+  /** a11y text for getting meaningful value. `$rating` and `$maxRating` are template variables and will be replaces by their corresponding properties.  */
+  @Prop() ariaLabelTranslation = '$rating out of $maxRating stars';
+  /** (optional) rating label */
+  @Prop({ reflect: true }) label?: string;
+  /** switch to define behaviour onTouch */
+  @State() firstStarSelected: boolean;
+
+  /** Emitted when the rating has changed */
+  @Event() scaleChange: EventEmitter;
 
-  renderIcon = (color: string, size: string, selected?: boolean) => {
-    if (selected) {
-      return `<scale-icon-action-favorite color=${color} size=${size} selected />`;
-    } else {
-      return `<scale-icon-action-favorite color=${color} size=${size} />`;
-    }
-  };
   connectedCallback() {
-    statusNote({ source: this.hostElement, tag: 'beta' });
-    this.handleMouseEnter = this.handleMouseEnter.bind(this);
-    this.handleMouseLeave = this.handleMouseLeave.bind(this);
-    this.handleMouseMove = this.handleMouseMove.bind(this);
-    this.handleMouseClick = this.handleMouseClick.bind(this);
-    this.handleKeyDown = this.handleKeyDown.bind(this);
-    this.handleTouchStart = this.handleTouchStart.bind(this);
-    this.handleTouchEnd = this.handleTouchEnd.bind(this);
-    this.handleTouchMove = this.handleTouchMove.bind(this);
+    this.firstStarSelected = false;
```

Comment:
if not intended, this might trigger an unnecessary render. I would set the default value while declaring it above, no?

```ts
@State() firstStarSelected: boolean = false;
```

Reply:
I have changed firstStarSelected to be a private property with a default value of false